src/sql/src/func.rs

```diff
@@ -3314,52 +3314,67 @@ pub static MZ_INTERNAL_BUILTINS: Lazy<BTreeMap<&'static str, Func>> = Lazy::new(
                         FROM
                         mz_internal.mz_object_dependencies AS d
                             JOIN
-                            mz_internal.mz_name_to_global_id($1) AS g (id)
+                            mz_internal.mz_resolve_object_name($1) AS g
                             ON d.object_id = g.id
                     ) AS d
                     ON s.id = d.subsource
             ") => ReturnType::set_of(RecordAny), oid::FUNC_MZ_GET_SUBSOURCES;
         },
-        "mz_name_to_global_id" => Table {
-            // If the user wants to specify a database, we also require them to specify which
-            // schemas in that database they want us to search in that database.
-            params!(String, ParamType::Plain(ScalarType::Array(Box::new(ScalarType::String))), String) => sql_impl_table_func("
-                SELECT o.id
+        "mz_resolve_object_name" => Table {
+            // This implementation is available primarily to drive the other, single-param
+            // implementation.
+            params!(
+                // Database
+                String,
+                // Schemas/search path
+                ParamType::Plain(ScalarType::Array(Box::new(ScalarType::String))),
+                // Item name
+                String
+            ) => sql_impl_table_func("
+                SELECT
+                    o.id, o.oid, o.schema_id, o.name, o.type, o.owner_id, o.privileges
                 FROM
                     mz_catalog.mz_objects AS o
                         JOIN mz_catalog.mz_schemas AS s ON o.schema_id = s.id
+                        JOIN unnest($2::text[]) AS search_schema (name)
+                            ON search_schema.name = s.name
                 WHERE
-                    o.name = $3::pg_catalog.text AND s.name =ANY ($2::pg_catalog.text[])
+                    o.name = $3::pg_catalog.text
                         AND
                     (
                         s.database_id IS NULL
```

Comment:
I _think_ this needs to be something like `(s.database_id IS NULL AND $1 IS NULL)`, I'm not positive though. Otherwise we ignore the input database if the matched object is in an ambient schema.

Reply:
If so, the second part of the condition actually might be enough because I think that 
```
                        s.database_id
                        = (
                            SELECT id FROM mz_catalog.mz_databases
                            WHERE name = $1::pg_catalog.text
                        )
```
will evaluate to `true` when `$1` is null and `s.database_id` is null.

Reply:
Oh yeah, good point. This new version also looks correct to me.